sklearn/decomposition/__init__.py

```diff
@@ -4,19 +4,30 @@
 this module can be regarded as dimensionality reduction techniques.
 """
 
-from ._nmf import NMF, non_negative_factorization
-from ._pca import PCA
-from ._incremental_pca import IncrementalPCA
-from ._kernel_pca import KernelPCA
-from ._sparse_pca import SparsePCA, MiniBatchSparsePCA
-from ._truncated_svd import TruncatedSVD
-from ._fastica import FastICA, fastica
+# TODO: remove me in 0.24:
+# pre-cache the import of the deprecated module to avoid showing
```

Comment:
showing -> shadowing?

Reply:
but I would rather say

> pre-cache the import of the deprecated module so that import sklearn.decomposition.dict_learning returns the function as in 0.21, instead of the module

which I believe is the root of the problem